react-main/src/app/screens/Game/components/Board/index.js

```diff
@@ -1,35 +1,12 @@
 import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
 
-import { calculateWinner } from './utils';
 import Square from './components/Square';
 
 class Board extends Component {
-  state = {
-    squares: Array(9).fill(null),
-    isNext: true
-  };
-
-  handleClick = i => {
-    const { isNext, squares } = this.state;
-    const diffSquares = squares.slice();
-    if (calculateWinner(squares) || squares[i]) return;
-    diffSquares[i] = isNext ? 'X' : 'O';
-    this.setState({ squares: diffSquares, isNext: !this.state.isNext });
-  };
-
-  renderSquare = i => <Square value={this.state.squares[i]} onClick={() => this.handleClick(i)} />;
+  renderSquare = i => <Square value={this.props.squares[i]} onClick={() => this.props.onClick(i)} />;
```

Comment:
What if square received number={i} and onClick={this.props.onClick}? Then you would leave logic to Square